src/gradata/enhancements/rule_integrity.py

```
@@ -0,0 +1,292 @@
+logger = logging.getLogger("gradata.rule_integrity")
+
+# ---------------------------------------------------------------------------
+# Key Management
+# ---------------------------------------------------------------------------
+
+_SECRET_KEY: bytes | None = None
+
+
+def _get_secret_key() -> bytes | None:
+    """Load the signing key from environment or return None (unsigned mode)."""
+    global _SECRET_KEY
```

Ruff (0.15.9)

[warning] Using the global statement to update `_SECRET_KEY` is discouraged

(PLW0603)